src/ngx_http_lua_headers_in.c

```diff
@@ -698,6 +693,11 @@ ngx_http_lua_set_input_header(ngx_http_request_t *r, ngx_str_t key,
     }
 #endif
 
+    if (r->headers_out.status == 400 || r->headers_in.headers.last == NULL) {
```

Comment:
I had suggested `||`to catch other potential (non-`400`) cases where `headers.last` is `NULL`.

I think it's a good idea to always short circuit on a `400`, regardless of whether `headers.last` is `NULL`.